benchmarks/device_uvector/device_uvector_bench.cu

```diff
@@ -66,4 +74,110 @@ BENCHMARK(BM_ThrustVectorSizeConstruction)
   ->Range(10'000, 1'000'000'000)  // NOLINT
   ->Unit(benchmark::kMicrosecond);
 
+// simple kernel used to test concurrent execution.
+__global__ void kernel(int const* input, int* output, std::size_t num)
+{
+  for (auto i = blockDim.x * blockIdx.x + threadIdx.x; i < num; i += gridDim.x * blockDim.x) {
+    output[i] = input[i] * input[i];
+  }
+}
+
+using thrust_vector = thrust::device_vector<int32_t>;
+using rmm_vector    = rmm::device_vector<int32_t>;
+using rmm_uvector   = rmm::device_uvector<int32_t>;
+
+template <typename Vector>
+Vector make_vector(int num_elements, rmm::cuda_stream_view stream, bool zero_init = false)
+{
+  static_assert(std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector> or
+                  std::is_same_v<Vector, rmm_uvector>,
+                "unsupported vector type");
+  if constexpr (std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector>) {
+    return Vector(num_elements, 0);
+  } else if constexpr (std::is_same_v<Vector, rmm_uvector>) {
+    auto vec = Vector(num_elements, stream);
+    if (zero_init) {
+      cudaMemsetAsync(vec.data(), 0, num_elements * sizeof(int32_t), stream.value());
+    }
+    return vec;
+  }
+}
+
+template <typename Vector>
+int32_t* vector_data(Vector& vec)
+{
+  static_assert(std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector> or
+                  std::is_same_v<Vector, rmm_uvector>,
+                "unsupported vector type");
+  if constexpr (std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector>) {
+    return vec.data().get();
+  } else if constexpr (std::is_same_v<Vector, rmm_uvector>) {
+    return vec.data();
+  }
+}
+
+template <typename Vector>
+void vector_workflow(std::size_t num_elements,
+                     int num_blocks,
+                     int block_size,
+                     rmm::cuda_stream const& input_stream,
+                     std::vector<rmm::cuda_stream> const& streams)
+{
+  auto input = make_vector<Vector>(num_elements, input_stream, true);
+  input_stream.synchronize();
+  for (rmm::cuda_stream_view stream : streams) {
+    auto vec = make_vector<Vector>(num_elements, stream);
+    kernel<<<num_blocks, block_size, 0, stream.value()>>>(
+      vector_data(input), vector_data(vec), num_elements);
+  }
+
+  for (rmm::cuda_stream_view stream : streams) {
+    stream.synchronize();
+  }
```

Comment:
I wonder if recording an event in the `input_stream` and using `cudaStreamWaitEvent` in `streams` would be lower overhead and make the timing less noisy? Likewise record events in `streams` after `kernel` and issue `cudaStreamWaitEvent`s to `input_stream`? Basically forking from `input_stream` to `streams` then joining back again with events. 

The `cuda_event_timer` will take care of synchronizing the `input_stream` for each benchmark iteration. 

This is probably overkill, so feel free to ignore.

Reply:
I did consider it, but since I was making this for a GTC slide, I wanted to keep the code as short as possible.

Reply:
I tried this just now and it made the code way more complex and for some reason screwed up the results. Rather than figure that out, I reverted it.